eclair-core/src/main/scala/fr/acinq/eclair/blockchain/electrum/ElectrumClientPool.scala

```diff
@@ -134,7 +141,10 @@ class ElectrumClientPool(serverAddresses: Set[ElectrumServerAddress])(implicit v
         statusListeners.foreach(_ ! ElectrumClient.ElectrumReady(height, tip, remoteAddress))
         context.system.eventStream.publish(ElectrumClient.ElectrumReady(height, tip, remoteAddress))
         goto(Connected) using ConnectedData(connection, Map(connection -> (height, tip)))
-      case Some(d) if height >= d.blockHeight + 2L =>
+      case Some(d) if connection != d.master && height >= d.blockHeight + 2L =>
```

Comment:
nitpick: it makes more sense if the 3 new comment lines were after the existing ones instead of before